src/reporter/mod.rs

```diff
@@ -0,0 +1,277 @@
+            "zone_id".into(),
+            manifest.zone_id.unwrap_or_else(|| "".into()),
+        );
+        project_info.insert("target_type".into(), manifest.target_type.to_string());
+        project_info.insert(
+            "workers_dev".into(),
+            manifest.workers_dev.unwrap_or(false).to_string(),
+        );
+        if let Some(builder) = manifest.build {
+            project_info.insert("using_custom_build".into(), "true".into());
+
+            if let Some((command, _)) = builder.build_command() {
+                project_info.insert("custom_build_command".into(), command.into());
+            }
+
+            // TODO: encode the format's struct members in map field instead of only string literal
+            project_info.insert(
+                "upload_format".into(),
+                match builder.upload {
+                    settings::toml::UploadFormat::ServiceWorker { .. } => "service-worker".into(),
+                    settings::toml::UploadFormat::Modules { .. } => "modules".into(),
+                },
+            );
+        }
+
+        if let Some(routes) = manifest.routes {
+            project_info.insert("routes".into(), routes.join(","));
+        }
+
+        if let Some(route) = manifest.route {
+            project_info.insert("route".into(), route);
+        }
+
+        if let Some(usage_model) = manifest.usage_model {
+            project_info.insert("usage_model".into(), usage_model.as_ref().into());
+        }
+    }
+
+    project_info
+}
+
+// removes frames before wrangler takes over at the panic, reduces noise
+fn useful_frames() -> String {
+    let bt = Backtrace::new();
+    let frames = bt.frames();
+    let found_idx = frames.iter().position(|frame| {
+        for sym in frame.symbols() {
+            if let Some(name) = sym.name() {
+                return name.to_string() == PANIC_UNWIND_START_MARKER;
+            }
+        }
+
+        false
+    });
+    let skip_count = match found_idx {
+        Some(idx) => idx + 1,
+        None => 0,
+    };
+
+    let useful = frames
+        .iter()
+        .skip(skip_count)
+        .filter(|frame| {
+            for sym in frame.symbols() {
+                if let Some(name) = sym.name() {
+                    return !name.to_string().starts_with(BACKTRACE_PATH_PREFIX);
+                }
+            }
+
+            true
+        })
+        .cloned()
+        .collect::<Vec<BacktraceFrame>>();
+
+    format!("{:?}", Backtrace::from(useful)).trim().to_string()
+}
+
+// extracts the payload contents from the panic (e.g. panic!("this is the payload"))
+// TODO: consider other <T> for downcast_ref and handle
+fn panic_payload(panic_info: Option<&PanicInfo>) -> Option<String> {
+    if let Some(info) = panic_info {
+        return match info.payload().downcast_ref::<&str>() {
+            Some(s) => Some(s.to_string()),
+            None => None,
+        };
+    }
+
+    None
+}
+
+fn err_exit<S: AsRef<str>>(msg: S, code: i32) {
```

Comment:
You should make this return `!` type, which will avoid peppering in the extra `unreachable!`s in the code.